internal/template/template_test.go

```diff
@@ -91,6 +91,33 @@ func TestMergeData(t *testing.T) {
 	}
 }
 
+func TestCopyData(t *testing.T) {
+	t.Run("", func(t *testing.T) {
+		nested := map[string]interface{}{
+			"c": []int{1, 2, 3},
+		}
+		outer := map[string]interface{}{
+			"a": nested,
+		}
+
+		got, err := CopyData(outer)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		if diff := cmp.Diff(outer, got); diff != "" {
+			t.Errorf("CopyData destination differs (-want +got):\n%v", diff)
+		}
+
+		// modify nested object
+		nested["b"] = 3
+
+		// deep copy (got) and outer map should differ now
```

Comment:
I think what you did before to compare the address is more straightforward than this.